src/revlanguage/functions/phylogenetics/ratematrix/Func_ConvertRateMatrix.cpp

```diff
@@ -0,0 +1,124 @@
+#include "Func_ConvertRateMatrix.h"
+
+#include "RealPos.h"
+#include "RlDeterministicNode.h"
+#include "RlMixtureModel.h"
+#include "RlRateMatrix.h"
+#include "RlSimplex.h"
+#include "GenericFunction.h"
+#include "TypedDagNode.h"
+#include "Argument.h"
+#include "ArgumentRule.h"
+#include "ArgumentRules.h"
+#include "RbException.h"
+#include "RevVariable.h"
+#include "RlFunction.h"
+#include "RlBoolean.h"
+#include "Integer.h"
+#include "Simplex.h"
+#include "TypeSpec.h"
+#include "ModelVector.h"
+#include "RbVector.h"
+#include "RbVectorImpl.h"
+
+
+#include "DistributionChisq.h"
+#include "RbMathFunctions.h"
+#include "UnitMixtureModel.h"
+
+using std::vector;
+using std::unique_ptr;
+
+namespace Core = RevBayesCore;
+
+Core::UnitMixtureModel* ConvertRateMatrixFunc(const Core::RateGenerator& model)
+{
+    if (auto matrix = dynamic_cast<const Core::RateMatrix*>(&model))
+        return new Core::UnitMixtureModel(*matrix, matrix->getStationaryFrequencies());
+    else
+        throw RbException()<<"_RateMatrix2MixtureModel: model is not a RateMatrix!";
```

Comment:
The problem is that RateGenerator does not necessary have stationary frequencies.  Only RateMatrix does.

If you write `x = [ fnJC(4) ]` then apparently `x` has type `RateGenerator[]` and not `RateMatrix[]`.  Maybe this could be fixed, but I have not looked into it yet...

Reply:
Oh, actually the arguments DO enforce it.  So, the exception should not occur.  I can put a note in.
But in `Func_ConvertVectoRateMatrix` I am not sure how to enforce this .

Reply:
It does -- this is just to catch a "this shouldn't happen" case.